File: JavaScript Fundamentals/Mid Exams/Mid Exam 1/theLift.js
function solve(arr) {
    let waitingPeople = Number(arr.shift());
    let wagon = arr.join(' ').split(' ').map(x => Number(x));
    let firstCase = false;
    let secondCase = false;
    let freeSpace = 0;
    
    for (let place of wagon) {
        freeSpace += 4 - place;
    }
    
    for (let index = 0; index < wagon.length; index++) {
        while (wagon[index] < 4 && waitingPeople > 0) {
            waitingPeople--;
            wagon[index]++
            freeSpace--;
        }

        if (waitingPeople == 0 && freeSpace > 0) {
            firstCase  = true;
            break;
        } else if (waitingPeople > 0 && freeSpace == 0) {
            secondCase = true;
            break;
        } else if (waitingPeople == 0 && freeSpace == 0) {
            break;
        }
    }

    if (firstCase) {
        console.log('The lift has empty spots!');
    } else if (secondCase) {
        console.log(`There isn't enough space! ${waitingPeople} people in a queue!`);
    } 
    console.log(wagon.join(' '));
}

File: Python Fundamentals/Mid Exam problems/the_lift.py
waiting_ppl = int(input())
lift_state = list(map(int, input().split()))

for index in range(len(lift_state)):
    current_state = lift_state[index]
    ppl_to_join = 4 - current_state
    if waiting_ppl < ppl_to_join:
        ppl_to_join = waiting_ppl
    waiting_ppl -= ppl_to_join
    lift_state[index] += ppl_to_join

if waiting_ppl > 0:
    print(f"There isn't enough space! {waiting_ppl} people in a queue!")
    print(' '.join(list(map(str, lift_state))))

elif lift_state.count(4) == len(lift_state) and waiting_ppl == 0:
    print(' '.join(list(map(str, lift_state))))

else:
    print("The lift has empty spots!")
    print(' '.join(list(map(str, lift_state))))
